language-support/hs/bindings/src/DA/Ledger/Services/TransactionService.hs

```diff
@@ -17,7 +25,105 @@ import Network.GRPC.HighLevel.Generated
 import qualified Com.Digitalasset.Ledger.Api.V1.TransactionService as LL
 import qualified Data.Map as Map
 
--- TODO:: all the other RPCs
+getTransactions :: GetTransactionsRequest -> LedgerService (Stream Transaction)
+getTransactions req =
+    makeLedgerService $ \timeout config -> do
+    stream <- newStream
+    _ <- forkIO $
+        withGRPCClient config $ \client -> do
+            service <- LL.transactionServiceClient client
+            let LL.TransactionService {transactionServiceGetTransactions=rpc} = service
+            sendToStreamFlat timeout (lowerRequest req) f stream rpc
+    return stream
+    where f = raiseList raiseTransaction . LL.getTransactionsResponseTransactions
+
+getTransactionTrees :: GetTransactionsRequest -> LedgerService (Stream TransactionTree)
+getTransactionTrees req =
+    makeLedgerService $ \timeout config -> do
+    stream <- newStream
+    _ <- forkIO $
+        withGRPCClient config $ \client -> do
+            service <- LL.transactionServiceClient client
+            let LL.TransactionService {transactionServiceGetTransactionTrees=rpc} = service
+            sendToStreamFlat timeout (lowerRequest req) f stream rpc
+    return stream
+    where f = raiseList raiseTransactionTree . LL.getTransactionTreesResponseTransactions
+
+getTransactionByEventId :: LedgerId -> EventId -> [Party] -> LedgerService (Maybe TransactionTree)
```

Comment:
Maybe add a comment on when you will get `Nothing` (only if you get a not found message) vs when you get an exception (all other failure cases).

Reply:
I think more generally we need to work out how best to link from the haskell bindings to the documentation which originates in the .proto files
```
// Lookup a transaction by the ID of an event that appears within it.
  // Returns ``NOT_FOUND`` if no such transaction exists.
  rpc GetFlatTransactionByEventId (GetTransactionByEventIdRequest) returns (GetFlatTransactionResponse);
```

and to document our general scheme of mapping `NOT_FOUND` (at the grpc level) to the `Maybe` type for the haskell binding.